lib/customer.rb

```diff
@@ -0,0 +1,30 @@
+require 'csv'
+
+class Customer
+  attr_accessor :email, :address
+  attr_reader :id
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    all_customers = CSV.open("data/customers.csv", "r").map do |line|
+      Customer.new(line[0].to_i, line[1], "#{line[2]} #{line[3]} #{line[4]} #{line[5]}")
```

Comment:
I love this implementation of interpolating everything and putting this new instance in a collection made by `.map`!